debugger/src/debugger/configuration.rs

```diff
@@ -0,0 +1,228 @@
+            // Update the `program_binary` and validate that the file exists.
+            target_core_config.program_binary = match qualify_and_update_os_file_path(
+                self.cwd.clone(),
+                target_core_config.program_binary.as_ref(),
+            ) {
+                Ok(program_binary) => {
+                    if !program_binary.is_file() {
+                        return Err(DebuggerError::Other(anyhow!(
+                            "Invalid program binary file specified '{:?}'",
+                            program_binary
+                        )));
+                    }
+                    Some(program_binary)
+                }
+                Err(error) => {
+                    return Err(DebuggerError::Other(anyhow!(
+                            "Please use the `program-binary` option to specify an executable for this target core. {:?}", error
+                        )));
+                }
+            };
+            // Update the `svd_file` and validate that the file exists.
+            // If there is a problem with this file, warn the user and continue with the session.
+            target_core_config.svd_file = match qualify_and_update_os_file_path(
+                self.cwd.clone(),
+                target_core_config.svd_file.as_ref(),
+            ) {
+                Ok(svd_file) => {
+                    if !svd_file.is_file() {
+                        log::error!("SVD file {:?} not found.", svd_file);
+                        None
+                    } else {
+                        Some(svd_file)
+                    }
+                }
+                Err(error) => {
+                    // SVD file is not mandatory.
+                    log::debug!("SVD file not specified: {:?}", &error);
+                    None
+                }
+            };
+        }
+
+        Ok(())
+    }
+
+    /// Validate the new cwd, or else set it from the environment.
+    pub(crate) fn validate_and_update_cwd(&self) -> Result<Option<PathBuf>, DebuggerError> {
+        Ok(match &self.cwd {
+            Some(temp_path) => {
+                if temp_path.is_dir() {
+                    Some(temp_path.to_path_buf())
+                } else if let Ok(current_dir) = current_dir() {
+                    Some(current_dir)
+                } else {
+                    log::error!("Cannot use current working directory. Please check existence and permissions.");
+                    None
+                }
+            }
+            None => {
+                if let Ok(current_dir) = current_dir() {
+                    Some(current_dir)
+                } else {
+                    log::error!("Cannot use current working directory. Please check existence and permissions.");
+                    None
+                }
+            }
+        })
+    }
+}
+
+/// If the path to the program to be debugged is relative, we join if with the cwd.
+fn qualify_and_update_os_file_path(
```

Comment:
I'd rather call this along the lines `get_absolute_path` which to me tells more what it does.

Reply:
Done